### PR TITLE
WT-11092 Add clang-format 10.0.0 aarch64 binary to s_clang_format

### PR DESCRIPTION
**Pull request by ajmorton**
Add auto-downloading of `clang-format 10.0.0 aarch64` binaries in `s_clang_format`.
I've also done a bit of cleanup to `download_clang_format` and added a comment explaining how to add new binaries in future.

### CODE REVIEW

**Comment by ajmorton** [comment]
I've confirmed this script works for `aarch64-Linux` and `x86_64-Linux` on my workstations and will confirm `x86_64-Darwin` on my laptop once it's finished updating xcode.

**Review by etienneptl — APPROVED** [review]
Thanks for doing this!

File: dist/s_clang_format [hunk 1]
```diff
@@ -8,24 +8,38 @@ trap 'rm -rf $t' 0 1 2 3 13 15
...
+
+    # Adding more clang-format binaries requires uploading them to boxes.10gen
+    # You can either get the clang-format binary from the llvm releases page
+    # (https://github.com/llvm/llvm-project/releases) or compile clang-format yourself.
+    # Place the binary in dist/ and confirm that s_clang_format runs correctly, then
+    # tar a folder containing just the clang-format binary with the format:
+    # clang-format-llvm-10.0.0-aarch64-linux-gnu/
+    #     clang-format
+    # This tarball should extract using the tar command below.
+    # The tarball can then be uploaded via a Jira request. See BUILD-17387 for an example.
```
Comment (lukech) [inline dist/s_clang_format:24]: I'd rather not mention the specific BUILD ticket. Whoever is interested should be able to git blame the WT ticket, which has a link to the BUILD ticket.

File: dist/s_clang_format [hunk 2]
```diff
@@ -8,24 +8,38 @@ trap 'rm -rf $t' 0 1 2 3 13 15
...
-        curl https://s3.amazonaws.com/boxes.10gen.com/build/clang-format-llvm-"$version"-x86_64-linux-gnu-ubuntu-20.04.tar-1.gz -o $archive
-        tar --strip=1 -C dist/ -xf $archive clang-format-llvm-"$version"-x86_64-linux-gnu-ubuntu-20.04/clang-format && rm $archive
-    elif [ "$sys_os" = "Darwin" ]; then
+
+    # Adding more clang-format binaries requires uploading them to boxes.10gen
+    # You can either get the clang-format binary from the llvm releases page
+    # (https://github.com/llvm/llvm-project/releases) or compile clang-format yourself.
+    # Place the binary in dist/ and confirm that s_clang_format runs correctly, then
+    # tar a folder containing just the clang-format binary with the format:
+    # clang-format-llvm-10.0.0-aarch64-linux-gnu/
```
Comment (lukech) [inline dist/s_clang_format:21]: Just to confirm this folder name is an example, that for other arch_and_os it should be named differently.

File: dist/s_clang_format [hunk 3]
```diff
@@ -8,24 +8,38 @@ trap 'rm -rf $t' 0 1 2 3 13 15
...
+    #     clang-format
+    # This tarball should extract using the tar command below.
+    # The tarball can then be uploaded via a Jira request. See BUILD-17387 for an example.
+    case $arch_and_os in
```
Comment (lukech) [inline dist/s_clang_format:25]: `uname -m` gave me `arm64` on my macbook. Should we cover this case as well?

Comment (lukech) [inline dist/s_clang_format:25]: `uname -m` gave me `arm64` on my macbook. Should we cover this case as well?

File: dist/s_clang_format [hunk 4]
```diff
@@ -8,24 +8,38 @@ trap 'rm -rf $t' 0 1 2 3 13 15
...
+    "aarch64-Linux")
+        curl https://s3.amazonaws.com/boxes.10gen.com/build/clang-format-llvm-"$version"-aarch64-linux-gnu.tar.gz -o $archive
+        tar --strip=1 -C dist/ -xf $archive clang-format-llvm-"$version"-aarch64-linux-gnu/clang-format && rm $archive
```
Comment (lukech) [inline dist/s_clang_format:28]: I'm wondering if we could standardize the tarball names to be e.g. `clang-format-llvm-"$version"-"$arch_and_os".tar.gz` and extract it to a folder named e.g. `clang-format-llvm-"$version"-"$arch_and_os"/clang-format` to save the `case in` logic.

File: dist/s_clang_format (hunk 1 above)
Comment (ajmorton) [reply dist/s_clang_format]: Good point. Fixed in [48d847b](https://github.com/wiredtiger/wiredtiger/pull/9181/commits/48d847b85b17698077d7b89faf68279f02ffc029)

File: dist/s_clang_format (hunk 2 above)
Comment (ajmorton) [reply dist/s_clang_format]: Yep this was an example. I've updated it to say `clang-format-llvm-${version}-${arch_and_os}.tar.gz` as per the other comment

File: dist/s_clang_format (hunk 3 above)
Comment (ajmorton) [reply dist/s_clang_format:26]: I'd prefer to leave this out of the current PR as this ticket unblocks our migration to the new ARM workstations.
I think it'd be useful to do alongside the WT-10619 upgrade to 12.0.1 though.

File: dist/s_clang_format (hunk 4 above)
Comment (ajmorton) [reply dist/s_clang_format:29]: Agreed on this. I think we can do this as part of WT-10619 also

File: dist/s_clang_format (hunk 3 above)
Comment (lukech) [reply dist/s_clang_format:26]: Fair enough

**Review by lukech — APPROVED** [review]
LGTM, thanks.

**Pull request merged**